### PR TITLE
Update DSL generator documentation

### LINKED ISSUE #172: How to use `GeneratedPathHelpersModule`?
Hi Tapioca team!

I recently learned about Tapioca from watching Shopify's Shipit! presentation on adopting Sorbet. I was very interested to try it out as my team and I have tried to adopt Sorbet on our large Rails projects in the past but have run into various complications.

First off, I'm happy to say that using Tapioca was much smoother than using Sorbet's built-in tooling! Adding it to a small project went almost entirely smoothly, whereas I could not get past the initial setup with the built-in tools. Thank you for this project!

I do have 1 question, however. In the [UrlHelpers](https://github.com/Shopify/tapioca/blob/master/manual/generator_urlhelpers.md) docs, it says a `GeneratedPathHelpersModule` is generated and included into the RBI for any class that includes the dynamic URL helpers module. I see that module is generated in an RBI file, and it is included in the RBI file for my classes.

My question then: How do I use the `GeneratedPathHelpersModule` module in my class?

Sorbet of course complains about the dynamic `include Rails.application.routes.url_helpers`, so I need to replace that with something.

But `GeneratedPathHelpersModule` only exists in the RBI files, so I cannot use `include GeneratedPathHelpersModule`. I get a NameError when I try that.

How do I replace `include Rails.application.routes.url_helpers`?

### PR DESCRIPTION
**Pull request by paracycle**
### Motivation

- Add comment note about safe include of UrlHelpers. Resolves #172.
- Turned naked URLs into Markdown links, so that the output is cleaner.
- Fixed some typos and wording mistakes.
- Fixed wrong usages of "Active Record".

### Implementation

See individual commits

### Tests

No tests.

### CODE REVIEW

File: lib/tapioca/compilers/dsl/active_resource.rb [hunk 1]
```diff
@@ -13,8 +13,9 @@ module Tapioca
   module Compilers
     module Dsl
       # `Tapioca::Compilers::Dsl::ActiveResource` decorates RBI files for subclasses of
-      # `ActiveResource::Base` which declare `schema` fields
-      # (see https://github.com/rails/activeresource).
+      # [`ActiveResource::Base`](https://github.com/rails/activeresource) which declare
+      # `schema` fields.
+      # .
```
Comment (Morriar) [inline lib/tapioca/compilers/dsl/active_resource.rb:18]: Why this dot?

File: lib/tapioca/compilers/dsl/active_support_current_attributes.rb [hunk 2]
```diff
@@ -13,9 +13,8 @@ module Tapioca
   module Compilers
     module Dsl
       # `Tapioca::Compilers::Dsl::ActiveSupportCurrentAttributes` decorates RBI files for all
-      # subclasses of `ActiveSupport::CurrentAttributes`
-      #
-      # To add attributes see https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html
+      # subclasses of
+      # [`ActiveSupport::CurrentAttributes`](https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html)
```
Comment (Morriar) [inline lib/tapioca/compilers/dsl/active_support_current_attributes.rb:17]:
```suggestion
      # [`ActiveSupport::CurrentAttributes`](https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html).
```

File: lib/tapioca/compilers/dsl/state_machines.rb [hunk 3]
```diff
@@ -15,10 +15,12 @@
...
     module Dsl
-      # `RbiGenerator::StateMachines` generates RBI files for classes that setup a `state_machine`
-      # (see https://github.com/state-machines/state_machines). The generator also processes the extra
-      # methods generated by [StateMachines Active Record](https://github.com/state-machines/state_machines-activerecord)
-      # and [StateMachines ActiveModel](https://github.com/state-machines/state_machines-activemodel) integrations.
+      # `Tapioca::Compilers::Dsl::StateMachines` generates RBI files for classes that setup a
+      # [`state_machine`](https://github.com/state-machines/state_machines). The generator also
+      # processes the extra methods generated by
+      # [StateMachines Active Record](https://github.com/state-machines/state_machines-activerecord)
+      # and [StateMachines ActiveModel](https://github.com/state-machines/state_machines-activemodel)
```
Comment (Morriar) [inline lib/tapioca/compilers/dsl/state_machines.rb:22]:
```suggestion
      # and [StateMachines Active Model](https://github.com/state-machines/state_machines-activemodel)
```

In https://guides.rubyonrails.org/active_model_basics.html they use a space like in `Active Record`.

File: lib/tapioca/compilers/dsl/url_helpers.rb [hunk 4]
```diff
@@ -32,7 +31,12 @@ module Dsl
       # ~~~rb
       # app/models/post.rb
       # class Post
-      #   include Rails.application.routes.url_helpers
+      #   # Use `T.unsafe` so that Sorbet does not complain about a dynamic
+      #   # module being included. This allows the `include` to happen properly
+      #   # at runtime but Sorbet won't see the include. However, since this
+      #   # generator will generate the proper RBI files for the include,
+      #   # type-checking will work as expected.
```
Comment (Morriar) [inline lib/tapioca/compilers/dsl/url_helpers.rb:38]:
```suggestion
      #   # static type checking will work as expected.
```

File: manual/generator_actioncontrollerhelpers.md [hunk 5]
```diff
@@ -1,8 +1,7 @@
 ## ActionControllerHelpers
 
 `Tapioca::Compilers::Dsl::ActionControllerHelpers` decorates RBI files for all
-subclasses of `::ActionController::Base`
-to add helper methods (see https://api.rubyonrails.org/classes/ActionController/Helpers.html).
+subclasses of [`ActionController::Base`](https://api.rubyonrails.org/classes/ActionController/Helpers.html)
```
Comment (Morriar) [inline manual/generator_actioncontrollerhelpers.md:4]:
```suggestion
subclasses of [`ActionController::Base`](https://api.rubyonrails.org/classes/ActionController/Helpers.html).
```

File: manual/generator_activerecordcolumns.md [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-`Tapioca::Compilers::Dsl::ActiveRecordColumns` refines RBI files for subclasses of `ActiveRecord::Base`
-(see https://api.rubyonrails.org/classes/ActiveRecord/Base.html). This generator is only
-responsible for defining the attribute methods that would be created for the columns that
-are defined in the Active Record model.
+`Tapioca::Compilers::Dsl::ActiveRecordColumns` refines RBI files for subclasses of
+[`ActiveRecord::Base`](https://api.rubyonrails.org/classes/ActiveRecord/Base.html).
+This generator is only responsible for defining the attribute methods that would be
+created for the columns that are defined in the Active Record model.
 
 **Note:** This generator, by default, generates weak signatures for column methods and treats each
```
Comment (Morriar) [inline manual/generator_activerecordcolumns.md:8]: Should we add an example of this?

File: manual/generator_activerecordscope.md [hunk 7]
```diff
@@ -1,8 +1,8 @@
 ## ActiveRecordScope
 
-`Tapioca::Compilers::Dsl::ActiveRecordScope` decorates RBI files for subclasses of
-`ActiveRecord::Base` which declare `scope` fields
-(see https://api.rubyonrails.org).
+`Tapioca::Compilers::Dsl::ActiveRecordScope` decorates RBI files for
+subclasses of `ActiveRecord::Base` which declare
+[`scope` fields](https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope)
```
Comment (Morriar) [inline manual/generator_activerecordscope.md:5]:
```suggestion
[`scope` fields](https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope).
```

File: manual/generator_activesupportcurrentattributes.md [hunk 8]
```diff
@@ -1,9 +1,8 @@
 ## ActiveSupportCurrentAttributes
 
 `Tapioca::Compilers::Dsl::ActiveSupportCurrentAttributes` decorates RBI files for all
-subclasses of `ActiveSupport::CurrentAttributes`
-
-To add attributes see https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html
+subclasses of
+[`ActiveSupport::CurrentAttributes`](https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html)
```
Comment (Morriar) [inline manual/generator_activesupportcurrentattributes.md:5]:
```suggestion
[`ActiveSupport::CurrentAttributes`](https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html).
```

File: manual/generator_statemachines.md [hunk 9]
```diff
@@ -1,9 +1,11 @@
...
 
-`RbiGenerator::StateMachines` generates RBI files for classes that setup a `state_machine`
-(see https://github.com/state-machines/state_machines). The generator also processes the extra
-methods generated by [StateMachines Active Record](https://github.com/state-machines/state_machines-activerecord)
-and [StateMachines ActiveModel](https://github.com/state-machines/state_machines-activemodel) integrations.
+`Tapioca::Compilers::Dsl::StateMachines` generates RBI files for classes that setup a
+[`state_machine`](https://github.com/state-machines/state_machines). The generator also
+processes the extra methods generated by
+[StateMachines Active Record](https://github.com/state-machines/state_machines-activerecord)
+and [StateMachines ActiveModel](https://github.com/state-machines/state_machines-activemodel)
```
Comment (Morriar) [inline manual/generator_statemachines.md:7]:
```suggestion
and [StateMachines Active Model](https://github.com/state-machines/state_machines-activemodel)
```

File: manual/generator_urlhelpers.md [hunk 10]
```diff
@@ -18,7 +17,12 @@ end
 ~~~rb
 app/models/post.rb
 class Post
-  include Rails.application.routes.url_helpers
+  # Use `T.unsafe` so that Sorbet does not complain about a dynamic
+  # module being included. This allows the `include` to happen properly
+  # at runtime but Sorbet won't see the include. However, since this
+  # generator will generate the proper RBI files for the include,
+  # type-checking will work as expected.
```
Comment (Morriar) [inline manual/generator_urlhelpers.md:24]:
```suggestion
  # static type checking will work as expected.
```

**Review by Morriar — APPROVED** [review]
Nice!

I suggested a few changes.

File: lib/tapioca/compilers/dsl/active_resource.rb (hunk 1 above)
Comment (paracycle) [reply lib/tapioca/compilers/dsl/active_resource.rb]: I'd missed it. Copy/paste error.

File: lib/tapioca/compilers/dsl/state_machines.rb (hunk 3 above)
Comment (paracycle) [reply lib/tapioca/compilers/dsl/state_machines.rb]: Good catch!

File: manual/generator_activerecordcolumns.md (hunk 6 above)
Comment (paracycle) [reply manual/generator_activerecordcolumns.md:8]: I think giving an example here will complicate the documentation much more. On the other hand, I think we can bundle a version of that module with Tapioca (or in another gem named Tapioca Runtime, or something): https://github.com/Shopify/tapioca/issues/174#issuecomment-736867460

**Pull request merged**